siriuspy/siriuspy/devices/orbit_interlock.py

```diff
@@ -101,25 +229,1349 @@ class BPMInterlock(_Device):
+
+        Returns:
+            intlk (numpy.ndarray, 160): interlock status for each BPM.
+        """
+        intlk = []
+        for bpm in self._devices:
+            intlk.append(bpm.trans_latch_bigger)
+        intlk = _np.array(intlk)
+        return intlk
+
+    @property
+    def trans_latch_smaller_x(self):
+        """Latch translation interlock set when X
+        minimum threshold is exceeded.
+
+        Returns:
+            intlk (numpy.ndarray, 160): interlock status for each BPM.
+        """
+        intlk = []
+        for bpm in self._devices:
+            intlk.append(bpm.trans_latch_smaller_x)
+        intlk = _np.array(intlk)
+        return intlk
+
+    @property
+    def trans_latch_bigger_x(self):
+        """Latch translation interlock set when X
+        maximum threshold is exceeded.
+
+        Returns:
+            intlk (numpy.ndarray, 160): interlock status for each BPM.
+        """
+        intlk = []
+        for bpm in self._devices:
+            intlk.append(bpm.trans_latch_bigger_x)
+        intlk = _np.array(intlk)
+        return intlk
+
+    @property
+    def trans_latch_smaller_y(self):
+        """Latch translation interlock set when Y
+        minimum threshold is exceeded.
+
+        Returns:
+            intlk (numpy.ndarray, 160): interlock status for each BPM.
+        """
+        intlk = []
+        for bpm in self._devices:
+            intlk.append(bpm.trans_latch_smaller_y)
+        intlk = _np.array(intlk)
+        return intlk
+
+    @property
+    def trans_latch_bigger_y(self):
+        """Latch translation interlock set when Y
+        maximum threshold is exceeded.
+
+        Returns:
+            intlk (numpy.ndarray, 160): interlock status for each BPM.
+        """
+        intlk = []
+        for bpm in self._devices:
+            intlk.append(bpm.trans_latch_bigger_y)
+        intlk = _np.array(intlk)
+        return intlk
+
+    #  --- angulation interlock ---
+
+    def cmd_ang_enable(self, timeout=TIMEOUT):
+        """Enable all BPM angulation interlock."""
+        for dev in self.devices:
+            dev.ang_enable = 1
+        return self._wait_devices_propty(
+            self.devices, 'IntlkAngEn-Sts', 1, timeout=timeout)
+
+    def cmd_ang_disable(self, timeout=TIMEOUT):
+        """Disable all BPM angulation interlock."""
+        for dev in self.devices:
+            dev.ang_enable = 0
+        return self._wait_devices_propty(
+            self.devices, 'IntlkAngEn-Sts', 0, timeout=timeout)
+
+    def cmd_reset_ang(self):
+        """Reset all BPM angulation interlock."""
+        for dev in self.devices:
+            dev.cmd_reset_ang()
+        return True
+
+    @property
+    def ang_thresminx(self):
+        """Minimum x angulation thresholds.
+
+        Returns:
+            thres (numpy.ndarray, 160):
+                min. x angulation threshold for each BPM.
+        """
+        thres = []
+        for bpm in self._devices:
```

Comment:
doing `thres = [b.ang_thresminx for b in self._devices]` is more pythonic.

The same is also valid for the other properties.

Reply:
done!